test/core/test/injector-mock.test.ts

```diff
@@ -67,7 +72,1149 @@ test('correctly mocks namespaced', () => {
+    expect(
+      await hoistSimpleCodeWithoutMocks(`
+import { test } from './test.js'
+
+for (const test of tests) {
+  console.log(test)
+}
+
+for (let test = 0; test < 10; test++) {
+  console.log(test)
+}
+
+for (const test in tests) {
+  console.log(test)
+}`),
+    ).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      vi.mock('faker')
+      const __vi_import_0__ = await import('./test.js')
+
+
+
+
+      for (const test of tests) {
+        console.log(test)
+      }
+
+      for (let test = 0; test < 10; test++) {
+        console.log(test)
+      }
+
+      for (const test in tests) {
+        console.log(test)
+      }
+      ;"
+    `)
+  })
+
+  test('avoid binding ClassExpression', async () => {
+    const result = await hoistSimpleCodeWithoutMocks(
+    `
+import Foo, { Bar } from './foo';
+
+console.log(Foo, Bar);
+const obj = {
+  foo: class Foo {},
+  bar: class Bar {}
+}
+const Baz = class extends Foo {}
+`,
+    )
+    expect(result).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      vi.mock('faker')
+      const __vi_import_0__ = await import('./foo')
+
+
+
+
+      console.log(__vi_import_0__.default, __vi_import_0__.Bar);
+      const obj = {
+        foo: class Foo {},
+        bar: class Bar {}
+      }
+      const Baz = class extends __vi_import_0__.default {}
+
+      ;"
+    `)
+  })
+
+  test('import assertion attribute', async () => {
+    expect(
+      await hoistSimpleCodeWithoutMocks(`
+  import * as foo from './foo.json' with { type: 'json' };
+  import('./bar.json', { with: { type: 'json' } });
+  `),
+    ).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      vi.mock('faker')
+      const __vi_import_0__ = await import('./foo.json')
+
+
+        
+        import('./bar.json', { with: { type: 'json' } });
+        
+      ;"
+    `)
+  })
+
+  test('import and export ordering', async () => {
+  // Given all imported modules logs `mod ${mod}` on execution,
+  // and `foo` is `bar`, the logging order should be:
+  // "mod a", "mod foo", "mod b", "bar1", "bar2"
+    expect(
+      await hoistSimpleCodeWithoutMocks(`
+console.log(foo + 1)
+export * from './a'
+import { foo } from './foo'
+export * from './b'
+console.log(foo + 2)
+  `),
+    ).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      vi.mock('faker')
+      const __vi_import_0__ = await import('./foo')
+
+
+      console.log(__vi_import_0__.foo + 1)
+      export * from './a'
+
+      export * from './b'
+      console.log(__vi_import_0__.foo + 2)
+        
+      ;"
+    `)
+  })
+
+  test('test', async () => {
+    expect(
+      await hoistSimpleCodeWithoutMocks(`
+      import { vi } from 'vitest'
+      import add, * as AddModule from '../src/add'
+      vi.mock('../src/add', () => {})`),
+    ).toMatchInlineSnapshot(`
+      "const { vi } = await import('vitest')
+      const { vi } = await import('vitest')
```

Comment:
This test is a bit weird. It was here before the refactor and it tested that the default export was correctly extracted

Right now, since `add` and `AddModule` are not used anywhere it looks redundant. Also the name "test" for a test is not telling much

Reply:
I'm sorry, I added this test case to test certain scenarios. I forgot to remove it.